runtime/src/main/java/app/cash/paraphrase/AndroidDateTimeConverter.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2023 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.paraphrase
+
+import android.icu.util.Calendar
+import android.icu.util.TimeZone
+import android.icu.util.ULocale
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+
+/**
+ * Converts `java.time` types used by Paraphrase to a [Calendar] that can be used by ICU to format.
+ */
+public object AndroidDateTimeConverter : DateTimeConverter<Calendar> {
+
+  private val Iso8601Locale by lazy(LazyThreadSafetyMode.NONE) {
+    ULocale.Builder()
+      .setExtension('u', "ca-iso8601")
+      .build()
+  }
```

Comment:
Since `AndroidDateTimeConverter` is an `object`, this `ULocale` instantiation was happening immediately when `FormattedResources` was accessed, causing the JVM test to (still) crash. Loading it lazily avoids this.